brave/brave-core
# Make top sites work like Chromium

Top sites on new tab page now work identical to how they work in Chromium. Custom logic was ditched in favor of using the same service that `chrome://new-tab-page` uses (`InstantService`). You can actually visit `chrome://new-tab-page` and you'll see the same tiles

Work left before this is complete:
- [x] Remove dead code
- [x] Translation string for custom links enabled (in settings)
- [x] Fix styling on tile hover (X should be in top right; doesn't need full banner)
- [x] Update this code to backfill the Chromium top sites visible setting (deprecate our version)
- [x] Fix super referrer tiles
- [x] Hide the default site that shows up (Chrome Web Store)
- [x] Fix tiles in our NTP are not updated w/o revisiting/reloading brave://new-tab-page
- [x] Verify if "Most visited" sites are movable in Chromium (this PR treats as non-movable)
- [x] Make sure default value for "customize top sites" is set properly
- [x] Don't persist the tile state to disk; clear when it goes out of scope and always load up empty

Fixes https://github.com/brave/brave-browser/issues/11500
Fixes https://github.com/brave/brave-browser/issues/9788
Fixes https://github.com/brave/brave-browser/issues/9457
Fixes https://github.com/brave/brave-browser/issues/11551

## Submitter Checklist:

- Verified that these changes build without errors on
- Verified that these changes pass automated tests (unit, browser, security tests) on

## Test Plan:
### Verify top site tiles are added
1. Fresh profile
2. Visit 5 - 10 websites
3. Verify sites are being added to new tab as you visit them
4. Open brave://new-tab-page and verify the sites basically match

### Re-ordering top site tiles
1. Follow steps for `Verify top site tiles are added`
2. In one window, have a Brave new tab. In the other, have brave://new-tab-page
3. In the Brave new tab, re-order the top sites (drag around, change order)
4. Verify the order changes on both pages
5. Close the browser and re-open
6. Verify the tiles are the same (order hasn't changed)
7. Visit more sites- verify sites are not being moved around. Since you have touched the tiles, they should stay there forever

### Removing sites and `Most Visited` mode
1. Follow steps for `Re-ordering top site tiles`
2. Remove a few sites
3. Close browser and re-open
4. Make sure changes were kept (removed tiles shouldn't come back)
5. Open new tab settings and disable `Customize top sites` setting
6. Top sites are now in `Most visited` mode. The tiles will have changed to ones which have been visited most frequently / most relevant.
7. Try to re-order the sites. Verify you CAN NOT re-order when `Customize top sites` is disabled

### Removing and restoring
1. Follow steps for `Verify top site tiles are added`
2. Remove a site
3. While the notice is showing, click `Undo` and ensure site is added back
4. Remove two sites this time
5. While the notice is showing, click `Restore all`
6. Verify all sites are restored

### Super Referral behavior
1. Fresh profile; put the text file in place to simulate SR download
2. Launch into Brave
3. Verify the top site tiles for that SR are showing
4. Verify that none of the SR tiles can be moved or deleted
5. Visit a few websites
6. Verify you can move the non-SR tiles
7. Disable SR via theme settings
8. Verify that SR tiles disappear

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on.
